metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -702,11 +745,15 @@ def _init_tableau_project_registry(self, all_project_map: dict) -> None:
 
         for project in list_of_skip_projects:
             if (
-                project.parent_id in self.tableau_project_registry
+                project.parent_id in projects_to_ingest
                 and self._is_denied_project(project) is False
             ):
                 logger.debug(f"Project {project.name} is added in project registry")
-                self.tableau_project_registry[project.id] = project
+                projects_to_ingest[project.id] = project
+
+        # We rely on automatic browse paths (v2) when creating containers. That's why we need to sort the projects here.
+        # Otherwise, nested projects will not have the correct browse paths if not created in correct order / hierarchy.
+        self.tableau_project_registry = OrderedDict(sorted(projects_to_ingest.items(), key=lambda item: len(item[1].path)))
```

Comment:
This was a bug that was already there before, when dealing with nested projects. If nested projects were added before their parent the browse path (v2) would be incorrect (wrong hierarchy). Let me know if there is a better way to do this.

Reply:
This works great, and is a good catch